### PR TITLE
Added SNI ( https://en.wikipedia.org/wiki/Server_Name_Indication ) su…

### PR DESCRIPTION
**Pull request by slaff**
…pport.

### CODE REVIEW

**Comment by igrr** [comment]
I think we also need to add a public method to set hostname, something like

``` c++
EXP_FUNC bool STDCALL ssl_set_hostname(const SSL *ssl, const char* hostname);
```

which would validate the length of `hostname` and call `strlcpy` to fill the internal structure.

**Comment by slaff** [comment]
@igrr I added the  `ssl_set_hostname` function. Check it when you have time.

File: ssl/tls1.h [hunk 1]
```diff
@@ -198,6 +198,7 @@ struct _SSL
     uint8_t read_sequence[8];       /* 64 bit sequence number */
     uint8_t write_sequence[8];      /* 64 bit sequence number */
     uint8_t hmac_header[SSL_RECORD_SIZE];    /* rx hmac */
+    const char host_name[255]; /* Needed for the SNI support */
```
Comment (Links2004) [inline ssl/tls1.h]: is it possible to use malloc for this.
to keep the memory profile low when SNI is not used or the host name is shorter?

**Comment by slaff** [comment]
@Links2004 Very good suggestion. Check the latest commit.

**Pull request merged**